packages/nimble-components/src/table/index.ts

```diff
@@ -161,6 +212,60 @@ export class Table<
 
         this.updateTableOptions({ columns: generatedColumns });
     }
+
```

Comment:
@jattasNI my [latest update](https://github.com/ni/nimble/pull/966/commits/e6b59ab1bd9c927a50389edb1067a9df610c64d8) has a quick attempt at separating out the virtualization logic into another class (could probably be improved but it's a starting point for discussion).

One thing to note (FYI @rajsite @atmgrifter00 ) is that this version has bindings in the template to `observable` properties not directly on the table class, but on an object instance that it owns (`(x: Table) => x.virtualizationHelper.someProperty` here), and that seems to work fine (table re-renders as those properties change). So if we were concerned about ending up with too many `public` "internal" properties being added to the table solely to bind to from the template, this might be one way we could avoid having so many / encapsulate them somewhat.  
Anything we need we assign with `ref` in the template has to stay on the table though (`viewport` in this PR).

Reply:
( @jattasNI @rajsite reset you to look at that update)

Reply:
I personally like the way it turned out. The TS code seems nicely encapsulated. The template code is a bit more verbose because of the need to drill into the`virtualizationHelper` a lot but over all I'd call it an improvement.